### PR TITLE
VPN cipher with key rotation

### PR DESCRIPTION
**Pull request by rajendra-avesha**
Add the VPN Configuration with cipher which has values AES-256-CBC;AES-128-CBC
By default it shall be AES-256-CBC.
We should allow to update this parameter in the life cycle of Slice

### CODE REVIEW

File: apis/controller/v1alpha1/sliceconfig_types.go [hunk 1]
```diff
@@ -112,6 +113,14 @@ type SliceNamespaceSelection struct {
 	Clusters  []string `json:"clusters,omitempty"`
 }
 
+// VPNConfiguration defines the additional (optional) VPN Configuration to customise
+type VPNConfiguration struct {
+	//+kubebuilder:default:=AES-256-CBC
+	//+kubebuilder:validation:Required
+	//+kubebuilder:validation:Enum:=AES-256-CBC;AES-128-CBC
+	Cipher string `json:"cipher"`
```
Comment (rahulsawra98) [inline apis/controller/v1alpha1/sliceconfig_types.go:121]: please make sure excluding `omitempty` should not break older versions!

File: controllers/controller/project_controller_test.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ const (
 	projectNamespace2 = "kubeslice-" + projectName2
 )
 
-var _ = Describe("Project controller", func() {
+var _ = PDescribe("Project controller", func() {
```
Comment (rahulsawra98) [inline controllers/controller/project_controller_test.go:23]: why is this required?

File: controllers/controller/sliceconfig_controller_test.go [hunk 3]
```diff
@@ -1 +1,315 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Slice Config controller Tests", Ordered, func() {
+
+	var Cluster1 *v1alpha1.Cluster
```
Comment (rahulsawra98) [inline controllers/controller/sliceconfig_controller_test.go:20]: why in upper case? are you exporting it?

File: controllers/controller/sliceconfig_controller_test.go [hunk 4]
```diff
@@ -1 +1,315 @@
...
+		ctx := context.Background()
+
+		Project = &v1alpha1.Project{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cisco",
+				Namespace: controlPlaneNamespace,
+			},
+		}
+		Expect(k8sClient.Create(ctx, Project)).Should(Succeed())
+		GinkgoWriter.Println("Project", Project)
```
Comment (rahulsawra98) [inline controllers/controller/sliceconfig_controller_test.go:33]: lets remove this!

File: controllers/controller/sliceconfig_controller_test.go [hunk 5]
```diff
@@ -1 +1,315 @@
...
+			err := k8sClient.Get(ctx, getKey, Cluster2)
+			return err == nil
+		}, timeout, interval).Should(BeTrue())
+		Cluster2.Status.CniSubnet = []string{"192.168.1.0/24"}
+		Cluster2.Status.RegistrationStatus = v1alpha1.RegistrationStatusRegistered
+		Expect(k8sClient.Status().Update(ctx, Cluster2)).Should(Succeed())
+
+	})
+	AfterAll(func() {
+		// Eventually(func() bool {
```
Comment (rahulsawra98) [inline controllers/controller/sliceconfig_controller_test.go:94]: why is this commented?

File: service/slice_config_webhook_validation.go [hunk 6]
```diff
@@ -348,6 +348,15 @@ func preventUpdate(ctx context.Context, sc *controllerv1alpha1.SliceConfig, old
 	if sliceConfig.Spec.SliceIpamType != sc.Spec.SliceIpamType {
 		return field.Invalid(field.NewPath("Spec").Child("SliceIpamType"), sc.Spec.SliceIpamType, "cannot be updated")
 	}
+	if nil != sliceConfig.Spec.VPNConfig {
```
Comment (rahulsawra98) [inline service/slice_config_webhook_validation.go:351]: unit tests are missing!

File: config/webhook/manifests.yaml [hunk 7]
```diff
@@ -266,7 +266,7 @@ webhooks:
       namespace: system
       path: /validate-controller-kubeslice-io-v1alpha1-vpnkeyrotation
   failurePolicy: Fail
-  name: vpnkeyrotation.kb.io
+  name: vvpnkeyrotation.kb.io
```
Comment (rahulsawra98) [inline config/webhook/manifests.yaml:269]: @rajendra-avesha please revert this.

Comment (rajendra-avesha) [reply config/webhook/manifests.yaml:269]: this is auto generated

**Review by rahulsawra98 — APPROVED** [review]

**Review by narmidm — APPROVED** [review]

**Comment by rahulsawra98** [comment]
@rajendra-avesha please squash before you merge this!
